### PR TITLE
Set up dummy repo for code review exercise

### PR DESCRIPTION
**Pull request by Enning94**
### Exercise: JS Best Practices- code review

in this exercise, I:

- created a new repository 
- I copied code from ToDo List project and committed to a new branch

### CODE REVIEW

File: src/modules/clearCheckedTasks.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const clearTodo = () => {
+  let todos;
```
Comment (dlwub) [inline src/modules/clearCheckedTasks.js:2]: The code:
`let todos; if (localStorage.getItem('todos') === null) { 
todos = []; } else {todos = JSON.parse(localStorage.getItem('todos')); 
}`  is repeated many times in your source code. 
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/displayTodo.js#L4-L9,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L2-L7,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L13-L18,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L32-L37,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L49-L54.
I think you could refactor it to a separate file and import it to where it is required.

Further, the above code could be made shorter if written as:
`let todos = []; 
if (localStorage.getItem('todos')) { 
todos = JSON.parse(localStorage.getItem('todos')); }`

File: src/modules/clearCheckedTasks.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  if (localStorage.getItem('todos') === null) {
+    todos = [];
+  } else {
+    todos = JSON.parse(localStorage.getItem('todos'));
+  }
```
Comment (dlwub) [inline src/modules/clearCheckedTasks.js:7]: The code:
`let todos; if (localStorage.getItem('todos') === null) { 
todos = []; } else {todos = JSON.parse(localStorage.getItem('todos')); 
}`  is repeated many times in your source code. 
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/displayTodo.js#L4-L9,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L2-L7,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L13-L18,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L32-L37,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L49-L54.
I think you could refactor it to a separate file and import it to where it is required.

Further, the above code could be made shorter if written as:
`let todos = []; 
if (localStorage.getItem('todos')) { 
todos = JSON.parse(localStorage.getItem('todos')); }`

File: src/modules/clearCheckedTasks.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  const uncompleted = todos.filter((todo) => todo.completed === false);
```
Comment (dlwub) [inline src/modules/clearCheckedTasks.js:8]: I think we can shorten this line of code as:
`const uncompleted = todos.filter((todo) => !todo.completed)`. What do you think?

**Review by dlwub — APPROVED** [review]
**Hi @Enning94:**
I appreciate your endeavor in the project. Great job in implementing ES6 syntax rules and following a modular approach. 

### Below you can find some optional comments:

File: src/modules/clearCheckedTasks.js (hunk 2 above)
Comment (dlwub) [inline src/modules/clearCheckedTasks.js:7]: The code:
`let todos; if (localStorage.getItem('todos') === null) { 
todos = []; } else {todos = JSON.parse(localStorage.getItem('todos')); 
}`  is repeated many times in your source code. 
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/displayTodo.js#L4-L9,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L2-L7,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L13-L18,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L32-L37,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L49-L54.
I think you could refactor it to a separate file and import it to where it is required.

Further, the above code could be made shorter if written as:
`let todos = []; 
if (localStorage.getItem('todos')) { 
todos = JSON.parse(localStorage.getItem('todos')); }`

File: src/modules/clearCheckedTasks.js (hunk 3 above)
Comment (dlwub) [inline src/modules/clearCheckedTasks.js:8]: I think we can shorten this line of code as:
`const uncompleted = todos.filter((todo) => !todo.completed)`. What do you think?

**Review by dlwub — APPROVED** [review]
**Hi @Enning94:**
I appreciate your endeavor in the project. Great job in implementing ES6 syntax rules and following a modular approach. 

### Below you can find some optional comments:

File: src/modules/clearCheckedTasks.js (hunk 2 above)
Comment (dlwub) [inline src/modules/clearCheckedTasks.js:7]: The code:
let todos; if (localStorage.getItem('todos') === null) { todos = []; } else {todos = JSON.parse(localStorage.getItem('todos')); } is repeated many times in your source code.
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/displayTodo.js#L4-L9,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L2-L7,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L13-L18,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L32-L37,
https://github.com/Enning94/JS-Best-Practices-ToDoList/blob/Feature/src/modules/localStorage.js#L49-L54.
I think you could refactor it to a separate file and import it to where it is required.

Further, the above code could be made shorter if written as:
let todos = []; if (localStorage.getItem('todos')) { todos = JSON.parse(localStorage.getItem('todos')); }

File: src/modules/clearCheckedTasks.js (hunk 3 above)
Comment (dlwub) [inline src/modules/clearCheckedTasks.js:8]: I think we can shorten this line of code as:
const uncompleted = todos.filter((todo) => !todo.completed). 
What do you think?

File: src/modules/clearCheckedTasks.js (hunk 2 above)
Comment (Enning94) [reply src/modules/clearCheckedTasks.js:7]: Sure, I will fix that. Thank you 👍

File: src/modules/clearCheckedTasks.js (hunk 3 above)
Comment (Enning94) [reply src/modules/clearCheckedTasks.js:8]: Alright, thank you.

**Comment by Enning94** [comment]
Hello Mr. Desta, I'm sorry I must send this email outside program hours.
I found it difficult to fix your review comments and messed up with the
repository at the end.
I couldn't revert the changes to the previous and I couldn't push new
changes either.
Because of that, I have created a new repo for the exercise. I will urge
you to review it so I can fix the suggestions and submit it before the
deadline.

This is  the link to the new repo:
https://github.com/Enning94/JS-Best-Practices/pull/1
Thank you
Festus

**Pull request merged**